[Automation] java run fails for com.clickhouse:jdbc-v2:0.9.7

## java run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=0.9.7 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=com.clickhouse:jdbc-v2:0.9.5`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27110323370
Last 300 lines of the log:
```console
SOFT-FAIL[native-image run][0.9.7][env GVM_TCK_LV=0.9.7 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=com.clickhouse:jdbc-v2:0.9.5]
========================================================================================
 com.clickhouse:jdbc-v2:0.9.5:0.9.7 stage javac compile
========================================================================================
Starting stage 'javac compile' for 0.9.7 at 2026-06-08T01:13:35Z
Command: env GVM_TCK_LV=0.9.7 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=com.clickhouse:jdbc-v2:0.9.5
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.clickhouse/jdbc-v2/0.9.5/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 849ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.clickhouse/jdbc-v2/0.9.5/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 0.9.7 at 2026-06-08T01:13:38Z with exit code 0 after 3s
========================================================================================
 com.clickhouse:jdbc-v2:0.9.5:0.9.7 stage java run
========================================================================================
Starting stage 'java run' for 0.9.7 at 2026-06-08T01:13:38Z
Command: env GVM_TCK_LV=0.9.7 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=com.clickhouse:jdbc-v2:0.9.5
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.clickhouse/jdbc-v2/0.9.5/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 820ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:jar UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava
> Task :processTestResources
> Task :testClasses

> Task :test FAILED

Jdbc_v2Test > statementMaintainsJdbcStateWithoutExecutingQueries() FAILED
    java.sql.SQLException: Failed to create connection
        at com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:127)
        at com.clickhouse.jdbc.Driver.connect(Driver.java:109)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.openConnection(Jdbc_v2Test.java:365)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.statementMaintainsJdbcStateWithoutExecutingQueries(Jdbc_v2Test.java:145)

        Caused by:
        com.clickhouse.client.api.ClientMisconfigurationException: Unknown and unmapped config properties: [disable_frameworks_detection]
            at app//com.clickhouse.client.api.ClientConfigProperties.parseConfigMap(ClientConfigProperties.java:372)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:148)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:113)
            at app//com.clickhouse.client.api.Client$Builder.build(Client.java:1154)
            at app//com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:106)
            ... 3 more

Jdbc_v2Test > connectionCreatesJdbcArraysFromClickHouseTypeNames() FAILED
    java.sql.SQLException: Failed to create connection
        at com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:127)
        at com.clickhouse.jdbc.Driver.connect(Driver.java:109)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.openConnection(Jdbc_v2Test.java:365)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.connectionCreatesJdbcArraysFromClickHouseTypeNames(Jdbc_v2Test.java:252)

        Caused by:
        com.clickhouse.client.api.ClientMisconfigurationException: Unknown and unmapped config properties: [disable_frameworks_detection]
            at app//com.clickhouse.client.api.ClientConfigProperties.parseConfigMap(ClientConfigProperties.java:372)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:148)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:113)
            at app//com.clickhouse.client.api.Client$Builder.build(Client.java:1154)
            at app//com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:106)
            ... 3 more

Jdbc_v2Test > dataSourceProvidesJdbcWrapperSemanticsAndConnections() FAILED
    java.sql.SQLException: Failed to create connection
        at com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:127)
        at com.clickhouse.jdbc.Driver.connect(Driver.java:109)
        at com.clickhouse.jdbc.DataSourceImpl.getConnection(DataSourceImpl.java:56)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.dataSourceProvidesJdbcWrapperSemanticsAndConnections(Jdbc_v2Test.java:99)

        Caused by:
        com.clickhouse.client.api.ClientMisconfigurationException: Unknown and unmapped config properties: [disable_frameworks_detection]
            at app//com.clickhouse.client.api.ClientConfigProperties.parseConfigMap(ClientConfigProperties.java:372)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:148)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:113)
            at app//com.clickhouse.client.api.Client$Builder.build(Client.java:1154)
            at app//com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:106)
            ... 3 more

Jdbc_v2Test > preparedStatementAcceptsCommonJavaValueTypes() FAILED
    java.sql.SQLException: Failed to create connection
        at com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:127)
        at com.clickhouse.jdbc.Driver.connect(Driver.java:109)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.openConnection(Jdbc_v2Test.java:365)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.preparedStatementAcceptsCommonJavaValueTypes(Jdbc_v2Test.java:227)

        Caused by:
        com.clickhouse.client.api.ClientMisconfigurationException: Unknown and unmapped config properties: [disable_frameworks_detection]
            at app//com.clickhouse.client.api.ClientConfigProperties.parseConfigMap(ClientConfigProperties.java:372)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:148)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:113)
            at app//com.clickhouse.client.api.Client$Builder.build(Client.java:1154)
            at app//com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:106)
            ... 3 more

Jdbc_v2Test > connectionExposesOfflineJdbcStateAndUnsupportedOperations() FAILED
    java.sql.SQLException: Failed to create connection
        at com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:127)
        at com.clickhouse.jdbc.Driver.connect(Driver.java:109)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.openConnection(Jdbc_v2Test.java:365)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.connectionExposesOfflineJdbcStateAndUnsupportedOperations(Jdbc_v2Test.java:107)

        Caused by:
        com.clickhouse.client.api.ClientMisconfigurationException: Unknown and unmapped config properties: [disable_frameworks_detection]
            at app//com.clickhouse.client.api.ClientConfigProperties.parseConfigMap(ClientConfigProperties.java:372)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:148)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:113)
            at app//com.clickhouse.client.api.Client$Builder.build(Client.java:1154)
            at app//com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:106)
            ... 3 more

Jdbc_v2Test > preparedStatementReportsParametersAndRejectsStatementSqlOverloads() FAILED
    java.sql.SQLException: Failed to create connection

12 tests completed, 6 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.clickhouse/jdbc-v2/0.9.5/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).
        at com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:127)
        at com.clickhouse.jdbc.Driver.connect(Driver.java:109)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.openConnection(Jdbc_v2Test.java:365)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.preparedStatementReportsParametersAndRejectsStatementSqlOverloads(Jdbc_v2Test.java:188)

        Caused by:
        com.clickhouse.client.api.ClientMisconfigurationException: Unknown and unmapped config properties: [disable_frameworks_detection]
            at app//com.clickhouse.client.api.ClientConfigProperties.parseConfigMap(ClientConfigProperties.java:372)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:148)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:113)
            at app//com.clickhouse.client.api.Client$Builder.build(Client.java:1154)
            at app//com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:106)
            ... 3 more

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.clickhouse/jdbc-v2/0.9.5/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 4 executed, 10 up-to-date

BUILD FAILED in 2s

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 0.9.7 at 2026-06-08T01:13:43Z with exit code 1 after 5s
❌ ERROR: 'java run' for 0.9.7 failed with exit code 1.
FAILED[java run][0.9.7][env GVM_TCK_LV=0.9.7 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=com.clickhouse:jdbc-v2:0.9.5]
```

## java run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=0.9.7 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=com.clickhouse:jdbc-v2:0.9.5`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27110323370
Last 300 lines of the log:
```console
SOFT-FAIL[native-image run][0.9.7][env GVM_TCK_LV=0.9.7 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=com.clickhouse:jdbc-v2:0.9.5]
========================================================================================
 com.clickhouse:jdbc-v2:0.9.5:0.9.7 stage javac compile
========================================================================================
Starting stage 'javac compile' for 0.9.7 at 2026-06-08T01:13:39Z
Command: env GVM_TCK_LV=0.9.7 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=com.clickhouse:jdbc-v2:0.9.5
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.clickhouse/jdbc-v2/0.9.5/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 855ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.clickhouse/jdbc-v2/0.9.5/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 0.9.7 at 2026-06-08T01:13:42Z with exit code 0 after 3s
========================================================================================
 com.clickhouse:jdbc-v2:0.9.5:0.9.7 stage java run
========================================================================================
Starting stage 'java run' for 0.9.7 at 2026-06-08T01:13:42Z
Command: env GVM_TCK_LV=0.9.7 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=com.clickhouse:jdbc-v2:0.9.5
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.clickhouse/jdbc-v2/0.9.5/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 793ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :processTestResources
> Task :compileTestJava
> Task :testClasses

> Task :test FAILED

Jdbc_v2Test > statementMaintainsJdbcStateWithoutExecutingQueries() FAILED
    java.sql.SQLException: Failed to create connection
        at com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:127)
        at com.clickhouse.jdbc.Driver.connect(Driver.java:109)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.openConnection(Jdbc_v2Test.java:365)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.statementMaintainsJdbcStateWithoutExecutingQueries(Jdbc_v2Test.java:145)

        Caused by:
        com.clickhouse.client.api.ClientMisconfigurationException: Unknown and unmapped config properties: [disable_frameworks_detection]
            at app//com.clickhouse.client.api.ClientConfigProperties.parseConfigMap(ClientConfigProperties.java:372)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:148)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:113)
            at app//com.clickhouse.client.api.Client$Builder.build(Client.java:1154)
            at app//com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:106)
            ... 3 more

Jdbc_v2Test > connectionCreatesJdbcArraysFromClickHouseTypeNames() FAILED
    java.sql.SQLException: Failed to create connection
        at com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:127)
        at com.clickhouse.jdbc.Driver.connect(Driver.java:109)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.openConnection(Jdbc_v2Test.java:365)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.connectionCreatesJdbcArraysFromClickHouseTypeNames(Jdbc_v2Test.java:252)

        Caused by:
        com.clickhouse.client.api.ClientMisconfigurationException: Unknown and unmapped config properties: [disable_frameworks_detection]
            at app//com.clickhouse.client.api.ClientConfigProperties.parseConfigMap(ClientConfigProperties.java:372)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:148)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:113)
            at app//com.clickhouse.client.api.Client$Builder.build(Client.java:1154)
            at app//com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:106)
            ... 3 more

Jdbc_v2Test > dataSourceProvidesJdbcWrapperSemanticsAndConnections() FAILED
    java.sql.SQLException: Failed to create connection
        at com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:127)
        at com.clickhouse.jdbc.Driver.connect(Driver.java:109)
        at com.clickhouse.jdbc.DataSourceImpl.getConnection(DataSourceImpl.java:56)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.dataSourceProvidesJdbcWrapperSemanticsAndConnections(Jdbc_v2Test.java:99)

        Caused by:
        com.clickhouse.client.api.ClientMisconfigurationException: Unknown and unmapped config properties: [disable_frameworks_detection]
            at app//com.clickhouse.client.api.ClientConfigProperties.parseConfigMap(ClientConfigProperties.java:372)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:148)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:113)
            at app//com.clickhouse.client.api.Client$Builder.build(Client.java:1154)
            at app//com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:106)
            ... 3 more

Jdbc_v2Test > preparedStatementAcceptsCommonJavaValueTypes() FAILED
    java.sql.SQLException: Failed to create connection
        at com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:127)
        at com.clickhouse.jdbc.Driver.connect(Driver.java:109)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.openConnection(Jdbc_v2Test.java:365)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.preparedStatementAcceptsCommonJavaValueTypes(Jdbc_v2Test.java:227)

        Caused by:
        com.clickhouse.client.api.ClientMisconfigurationException: Unknown and unmapped config properties: [disable_frameworks_detection]
            at app//com.clickhouse.client.api.ClientConfigProperties.parseConfigMap(ClientConfigProperties.java:372)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:148)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:113)
            at app//com.clickhouse.client.api.Client$Builder.build(Client.java:1154)
            at app//com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:106)
            ... 3 more

Jdbc_v2Test > connectionExposesOfflineJdbcStateAndUnsupportedOperations() FAILED
    java.sql.SQLException: Failed to create connection
        at com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:127)
        at com.clickhouse.jdbc.Driver.connect(Driver.java:109)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.openConnection(Jdbc_v2Test.java:365)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.connectionExposesOfflineJdbcStateAndUnsupportedOperations(Jdbc_v2Test.java:107)

        Caused by:
        com.clickhouse.client.api.ClientMisconfigurationException: Unknown and unmapped config properties: [disable_frameworks_detection]
            at app//com.clickhouse.client.api.ClientConfigProperties.parseConfigMap(ClientConfigProperties.java:372)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:148)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:113)
            at app//com.clickhouse.client.api.Client$Builder.build(Client.java:1154)
            at app//com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:106)
            ... 3 more

Jdbc_v2Test > preparedStatementReportsParametersAndRejectsStatementSqlOverloads() FAILED
    java.sql.SQLException: Failed to create connection
        at com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:127)
        at com.clickhouse.jdbc.Driver.connect(Driver.java:109)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.openConnection(Jdbc_v2Test.java:365)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.preparedStatementReportsParametersAndRejectsStatementSqlOverloads(Jdbc_v2Test.java:188)

        Caused by:
        com.clickhouse.client.api.ClientMisconfigurationException: Unknown and unmapped config properties: [disable_frameworks_detection]
            at app//com.clickhouse.client.api.ClientConfigProperties.parseConfigMap(ClientConfigProperties.java:372)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:148)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:113)
            at app//com.clickhouse.client.api.Client$Builder.build(Client.java:1154)
            at app//com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:106)
            ... 3 more

12 tests completed, 6 failed

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.clickhouse/jdbc-v2/0.9.5/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 4 executed, 10 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.clickhouse/jdbc-v2/0.9.5/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 0.9.7 at 2026-06-08T01:13:47Z with exit code 1 after 5s
❌ ERROR: 'java run' for 0.9.7 failed with exit code 1.
FAILED[java run][0.9.7][env GVM_TCK_LV=0.9.7 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=com.clickhouse:jdbc-v2:0.9.5]
```

## java run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `env GVM_TCK_LV=0.9.7 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean javaTest -Pcoordinates=com.clickhouse:jdbc-v2:0.9.5`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27110323370
Last 300 lines of the log:
```console
SOFT-FAIL[native-image run][0.9.7][env GVM_TCK_LV=0.9.7 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=com.clickhouse:jdbc-v2:0.9.5]
========================================================================================
 com.clickhouse:jdbc-v2:0.9.5:0.9.7 stage javac compile
========================================================================================
Starting stage 'javac compile' for 0.9.7 at 2026-06-08T01:13:50Z
Command: env GVM_TCK_LV=0.9.7 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=com.clickhouse:jdbc-v2:0.9.5
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.clickhouse/jdbc-v2/0.9.5/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 848ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.clickhouse/jdbc-v2/0.9.5/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 0.9.7 at 2026-06-08T01:13:53Z with exit code 0 after 3s
========================================================================================
 com.clickhouse:jdbc-v2:0.9.5:0.9.7 stage java run
========================================================================================
Starting stage 'java run' for 0.9.7 at 2026-06-08T01:13:53Z
Command: env GVM_TCK_LV=0.9.7 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean javaTest -Pcoordinates=com.clickhouse:jdbc-v2:0.9.5
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.clickhouse/jdbc-v2/0.9.5/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 821ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :processTestResources
> Task :compileTestJava
> Task :testClasses

> Task :test FAILED

Jdbc_v2Test > statementMaintainsJdbcStateWithoutExecutingQueries() FAILED
    java.sql.SQLException: Failed to create connection
        at com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:127)
        at com.clickhouse.jdbc.Driver.connect(Driver.java:109)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.openConnection(Jdbc_v2Test.java:365)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.statementMaintainsJdbcStateWithoutExecutingQueries(Jdbc_v2Test.java:145)

        Caused by:
        com.clickhouse.client.api.ClientMisconfigurationException: Unknown and unmapped config properties: [disable_frameworks_detection]
            at app//com.clickhouse.client.api.ClientConfigProperties.parseConfigMap(ClientConfigProperties.java:372)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:148)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:113)
            at app//com.clickhouse.client.api.Client$Builder.build(Client.java:1154)
            at app//com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:106)
            ... 3 more

Jdbc_v2Test > connectionCreatesJdbcArraysFromClickHouseTypeNames() FAILED
    java.sql.SQLException: Failed to create connection
        at com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:127)
        at com.clickhouse.jdbc.Driver.connect(Driver.java:109)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.openConnection(Jdbc_v2Test.java:365)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.connectionCreatesJdbcArraysFromClickHouseTypeNames(Jdbc_v2Test.java:252)

        Caused by:
        com.clickhouse.client.api.ClientMisconfigurationException: Unknown and unmapped config properties: [disable_frameworks_detection]
            at app//com.clickhouse.client.api.ClientConfigProperties.parseConfigMap(ClientConfigProperties.java:372)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:148)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:113)
            at app//com.clickhouse.client.api.Client$Builder.build(Client.java:1154)
            at app//com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:106)
            ... 3 more

Jdbc_v2Test > dataSourceProvidesJdbcWrapperSemanticsAndConnections() FAILED
    java.sql.SQLException: Failed to create connection
        at com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:127)
        at com.clickhouse.jdbc.Driver.connect(Driver.java:109)
        at com.clickhouse.jdbc.DataSourceImpl.getConnection(DataSourceImpl.java:56)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.dataSourceProvidesJdbcWrapperSemanticsAndConnections(Jdbc_v2Test.java:99)

        Caused by:
        com.clickhouse.client.api.ClientMisconfigurationException: Unknown and unmapped config properties: [disable_frameworks_detection]
            at app//com.clickhouse.client.api.ClientConfigProperties.parseConfigMap(ClientConfigProperties.java:372)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:148)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:113)
            at app//com.clickhouse.client.api.Client$Builder.build(Client.java:1154)
            at app//com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:106)
            ... 3 more


12 tests completed, 6 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.clickhouse/jdbc-v2/0.9.5/build/reports/tests/test/index.html

* Try:
> Run with Jdbc_v2Test > preparedStatementAcceptsCommonJavaValueTypes() FAILED
    java.sql.SQLException: Failed to create connection
        at com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:127)
        at com.clickhouse.jdbc.Driver.connect(Driver.java:109)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.openConnection(Jdbc_v2Test.java:365)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.preparedStatementAcceptsCommonJavaValueTypes(Jdbc_v2Test.java:227)

        Caused by:
        com.clickhouse.client.api.ClientMisconfigurationException: Unknown and unmapped config properties: [disable_frameworks_detection]
            at app//com.clickhouse.client.api.ClientConfigProperties.parseConfigMap(ClientConfigProperties.java:372)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:148)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:113)
            at app//com.clickhouse.client.api.Client$Builder.build(Client.java:1154)
            at app//com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:106)
            ... 3 more

Jdbc_v2Test > connectionExposesOfflineJdbcStateAndUnsupportedOperations() FAILED
    java.sql.SQLException: Failed to create connection
        at com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:127)
        at com.clickhouse.jdbc.Driver.connect(Driver.java:109)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.openConnection(Jdbc_v2Test.java:365)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.connectionExposesOfflineJdbcStateAndUnsupportedOperations(Jdbc_v2Test.java:107)

        Caused by:
        com.clickhouse.client.api.ClientMisconfigurationException: Unknown and unmapped config properties: [disable_frameworks_detection]
            at app//com.clickhouse.client.api.ClientConfigProperties.parseConfigMap(ClientConfigProperties.java:372)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:148)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:113)
            at app//com.clickhouse.client.api.Client$Builder.build(Client.java:1154)
            at app//com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:106)
            ... 3 more

Jdbc_v2Test > preparedStatementReportsParametersAndRejectsStatementSqlOverloads() FAILED
    java.sql.SQLException: Failed to create connection
        at com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:127)
        at com.clickhouse.jdbc.Driver.connect(Driver.java:109)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.openConnection(Jdbc_v2Test.java:365)
        at com_clickhouse.jdbc_v2.Jdbc_v2Test.preparedStatementReportsParametersAndRejectsStatementSqlOverloads(Jdbc_v2Test.java:188)

        Caused by:
        com.clickhouse.client.api.ClientMisconfigurationException: Unknown and unmapped config properties: [disable_frameworks_detection]
            at app//com.clickhouse.client.api.ClientConfigProperties.parseConfigMap(ClientConfigProperties.java:372)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:148)
            at app//com.clickhouse.client.api.Client.<init>(Client.java:113)
            at app//com.clickhouse.client.api.Client$Builder.build(Client.java:1154)
            at app//com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:106)
            ... 3 more

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.clickhouse/jdbc-v2/0.9.5/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 4 executed, 10 up-to-date
--scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 0.9.7 at 2026-06-08T01:13:58Z with exit code 1 after 5s
❌ ERROR: 'java run' for 0.9.7 failed with exit code 1.
FAILED[java run][0.9.7][env GVM_TCK_LV=0.9.7 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean javaTest -Pcoordinates=com.clickhouse:jdbc-v2:0.9.5]
```

--- a/metadata/com.clickhouse/jdbc-v2/0.9.7/reachability-metadata.json
+++ b/metadata/com.clickhouse/jdbc-v2/0.9.7/reachability-metadata.json
@@ -1,0 +1,132 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.client.api.Client"
+      },
+      "type": "byte[][]"
+    },
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.jdbc.internal.JdbcConfiguration"
+      },
+      "type": "java.lang.Object[]"
+    },
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.jdbc.internal.JdbcUtils"
+      },
+      "type": "java.lang.String[][]"
+    },
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.client.api.internal.HttpAPIClientHelper"
+      },
+      "type": "java.net.InetAddress[]"
+    },
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.jdbc.Driver"
+      },
+      "type": "java.sql.DriverPropertyInfo[]"
+    },
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.jdbc.StatementImpl"
+      },
+      "type": "java.sql.SQLException"
+    },
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.client.api.DataTypeUtils"
+      },
+      "type": "java.time.format.DateTimeFormatterBuilder$DateTimePrinterParser[]"
+    },
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.jdbc.internal.JdbcUtils"
+      },
+      "type": "java.util.Map$Entry[]"
+    },
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.client.api.internal.HttpAPIClientHelper"
+      },
+      "type": "jdk.net.LinuxSocketOptions"
+    },
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.client.api.Client"
+      },
+      "type": "net.jpountz.lz4.LZ4HCJNICompressor"
+    },
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.client.api.Client"
+      },
+      "type": "net.jpountz.lz4.LZ4JNICompressor"
+    },
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.client.api.Client"
+      },
+      "type": "net.jpountz.lz4.LZ4JNISafeDecompressor"
+    },
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.client.api.internal.HttpAPIClientHelper"
+      },
+      "type": "org.apache.hc.core5.http.Header[]"
+    },
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.client.api.Client"
+      },
+      "type": "sun.security.provider.NativePRNG"
+    },
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.client.api.Client"
+      },
+      "type": "sun.security.provider.SHA"
+    },
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.jdbc.ConnectionImpl"
+      },
+      "type": "sun.util.resources.cldr.CalendarData"
+    }
+  ],
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.logging.LoggerFactory"
+      },
+      "glob": "META-INF/services/com.clickhouse.logging.LoggerFactory"
+    },
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.jdbc.StatementImpl"
+      },
+      "glob": "META-INF/services/org.slf4j.spi.SLF4JServiceProvider"
+    },
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.client.config.ClickHouseClientOption"
+      },
+      "glob": "client-v2-version.properties"
+    },
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.client.config.ClickHouseClientOption"
+      },
+      "glob": "jdbc-v2-version.properties"
+    },
+    {
+      "condition": {
+        "typeReached": "com.clickhouse.client.api.internal.HttpAPIClientHelper"
+      },
+      "glob": "org/apache/hc/client5/version.properties"
+    }
+  ]
+}

--- a/metadata/com.clickhouse/jdbc-v2/0.9.7/reachability-metadata.json
+++ b/metadata/com.clickhouse/jdbc-v2/0.9.7/reachability-metadata.json
@@ -1,132 +1,132 @@
 {
-  "reflection": [
+  "reflection" : [
     {
-      "condition": {
-        "typeReached": "com.clickhouse.client.api.Client"
+      "condition" : {
+        "typeReached" : "com.clickhouse.client.api.Client"
       },
-      "type": "byte[][]"
+      "type" : "byte[][]"
     },
     {
-      "condition": {
-        "typeReached": "com.clickhouse.jdbc.internal.JdbcConfiguration"
+      "condition" : {
+        "typeReached" : "com.clickhouse.jdbc.internal.JdbcConfiguration"
       },
-      "type": "java.lang.Object[]"
+      "type" : "java.lang.Object[]"
     },
     {
-      "condition": {
-        "typeReached": "com.clickhouse.jdbc.internal.JdbcUtils"
+      "condition" : {
+        "typeReached" : "com.clickhouse.jdbc.internal.JdbcUtils"
       },
-      "type": "java.lang.String[][]"
+      "type" : "java.lang.String[][]"
     },
     {
-      "condition": {
-        "typeReached": "com.clickhouse.client.api.internal.HttpAPIClientHelper"
+      "condition" : {
+        "typeReached" : "com.clickhouse.client.api.internal.HttpAPIClientHelper"
       },
-      "type": "java.net.InetAddress[]"
+      "type" : "java.net.InetAddress[]"
     },
     {
-      "condition": {
-        "typeReached": "com.clickhouse.jdbc.Driver"
+      "condition" : {
+        "typeReached" : "com.clickhouse.jdbc.Driver"
       },
-      "type": "java.sql.DriverPropertyInfo[]"
+      "type" : "java.sql.DriverPropertyInfo[]"
     },
     {
-      "condition": {
-        "typeReached": "com.clickhouse.jdbc.StatementImpl"
+      "condition" : {
+        "typeReached" : "com.clickhouse.jdbc.StatementImpl"
       },
-      "type": "java.sql.SQLException"
+      "type" : "java.sql.SQLException"
     },
     {
-      "condition": {
-        "typeReached": "com.clickhouse.client.api.DataTypeUtils"
+      "condition" : {
+        "typeReached" : "com.clickhouse.client.api.DataTypeUtils"
       },
-      "type": "java.time.format.DateTimeFormatterBuilder$DateTimePrinterParser[]"
+      "type" : "java.time.format.DateTimeFormatterBuilder$DateTimePrinterParser[]"
     },
     {
-      "condition": {
-        "typeReached": "com.clickhouse.jdbc.internal.JdbcUtils"
+      "condition" : {
+        "typeReached" : "com.clickhouse.jdbc.internal.JdbcUtils"
       },
-      "type": "java.util.Map$Entry[]"
+      "type" : "java.util.Map$Entry[]"
     },
     {
-      "condition": {
-        "typeReached": "com.clickhouse.client.api.internal.HttpAPIClientHelper"
+      "condition" : {
+        "typeReached" : "com.clickhouse.client.api.internal.HttpAPIClientHelper"
       },
-      "type": "jdk.net.LinuxSocketOptions"
+      "type" : "jdk.net.LinuxSocketOptions"
     },
     {
-      "condition": {
-        "typeReached": "com.clickhouse.client.api.Client"
+      "condition" : {
+        "typeReached" : "com.clickhouse.client.api.Client"
       },
-      "type": "net.jpountz.lz4.LZ4HCJNICompressor"
+      "type" : "net.jpountz.lz4.LZ4HCJNICompressor"
     },
     {
-      "condition": {
-        "typeReached": "com.clickhouse.client.api.Client"
+      "condition" : {
+        "typeReached" : "com.clickhouse.client.api.Client"
       },
-      "type": "net.jpountz.lz4.LZ4JNICompressor"
+      "type" : "net.jpountz.lz4.LZ4JNICompressor"
     },
     {
-      "condition": {
-        "typeReached": "com.clickhouse.client.api.Client"
+      "condition" : {
+        "typeReached" : "com.clickhouse.client.api.Client"
       },
-      "type": "net.jpountz.lz4.LZ4JNISafeDecompressor"
+      "type" : "net.jpountz.lz4.LZ4JNISafeDecompressor"
     },
     {
-      "condition": {
-        "typeReached": "com.clickhouse.client.api.internal.HttpAPIClientHelper"
+      "condition" : {
+        "typeReached" : "com.clickhouse.client.api.internal.HttpAPIClientHelper"
       },
-      "type": "org.apache.hc.core5.http.Header[]"
+      "type" : "org.apache.hc.core5.http.Header[]"
     },
     {
-      "condition": {
-        "typeReached": "com.clickhouse.client.api.Client"
+      "condition" : {
+        "typeReached" : "com.clickhouse.client.api.Client"
       },
-      "type": "sun.security.provider.NativePRNG"
+      "type" : "sun.security.provider.NativePRNG"
     },
     {
-      "condition": {
-        "typeReached": "com.clickhouse.client.api.Client"
+      "condition" : {
+        "typeReached" : "com.clickhouse.client.api.Client"
       },
-      "type": "sun.security.provider.SHA"
+      "type" : "sun.security.provider.SHA"
     },
     {
-      "condition": {
-        "typeReached": "com.clickhouse.jdbc.ConnectionImpl"
+      "condition" : {
+        "typeReached" : "com.clickhouse.jdbc.ConnectionImpl"
       },
-      "type": "sun.util.resources.cldr.CalendarData"
+      "type" : "sun.util.resources.cldr.CalendarData"
     }
   ],
-  "resources": [
+  "resources" : [
     {
-      "condition": {
-        "typeReached": "com.clickhouse.logging.LoggerFactory"
+      "condition" : {
+        "typeReached" : "com.clickhouse.logging.LoggerFactory"
       },
-      "glob": "META-INF/services/com.clickhouse.logging.LoggerFactory"
+      "glob" : "META-INF/services/com.clickhouse.logging.LoggerFactory"
     },
     {
-      "condition": {
-        "typeReached": "com.clickhouse.jdbc.StatementImpl"
+      "condition" : {
+        "typeReached" : "com.clickhouse.jdbc.StatementImpl"
       },
-      "glob": "META-INF/services/org.slf4j.spi.SLF4JServiceProvider"
+      "glob" : "META-INF/services/org.slf4j.spi.SLF4JServiceProvider"
     },
     {
-      "condition": {
-        "typeReached": "com.clickhouse.client.config.ClickHouseClientOption"
+      "condition" : {
+        "typeReached" : "com.clickhouse.client.config.ClickHouseClientOption"
       },
-      "glob": "client-v2-version.properties"
+      "glob" : "client-v2-version.properties"
     },
     {
-      "condition": {
-        "typeReached": "com.clickhouse.client.config.ClickHouseClientOption"
+      "condition" : {
+        "typeReached" : "com.clickhouse.client.config.ClickHouseClientOption"
       },
-      "glob": "jdbc-v2-version.properties"
+      "glob" : "jdbc-v2-version.properties"
     },
     {
-      "condition": {
-        "typeReached": "com.clickhouse.client.api.internal.HttpAPIClientHelper"
+      "condition" : {
+        "typeReached" : "com.clickhouse.client.api.internal.HttpAPIClientHelper"
       },
-      "glob": "org/apache/hc/client5/version.properties"
+      "glob" : "org/apache/hc/client5/version.properties"
     }
   ]
 }

--- a/metadata/com.clickhouse/jdbc-v2/index.json
+++ b/metadata/com.clickhouse/jdbc-v2/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "0.9.7",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/com/clickhouse/jdbc-v2/$version$/jdbc-v2-$version$-sources.jar",
+    "repository-url" : "https://github.com/ClickHouse/clickhouse-java",
+    "test-code-url" : "https://github.com/ClickHouse/clickhouse-java/tree/v$version$/jdbc-v2/src/test",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/com/clickhouse/jdbc-v2/$version$/jdbc-v2-$version$-javadoc.jar",
+    "description" : "ClickHouse JDBC Driver V2 implements the standard Java Database Connectivity API for connecting JVM applications to ClickHouse databases. It builds on the ClickHouse Java client to execute SQL statements, read result sets, handle metadata, and support JDBC DataSource and Driver integrations.",
     "tested-versions" : [
       "0.9.7"
     ],

--- a/metadata/com.clickhouse/jdbc-v2/index.json
+++ b/metadata/com.clickhouse/jdbc-v2/index.json
@@ -1,6 +1,15 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "0.9.7",
+    "tested-versions" : [
+      "0.9.7"
+    ],
+    "allowed-packages" : [
+      "com.clickhouse"
+    ]
+  },
+  {
     "metadata-version" : "0.9.5",
     "source-code-url" : "https://repo.maven.apache.org/maven2/com/clickhouse/jdbc-v2/$version$/jdbc-v2-$version$-sources.jar",
     "repository-url" : "https://github.com/ClickHouse/clickhouse-java",

--- a/stats/com.clickhouse/jdbc-v2/0.9.7/execution-metrics.json
+++ b/stats/com.clickhouse/jdbc-v2/0.9.7/execution-metrics.json
@@ -1,0 +1,101 @@
+{
+  "fix_java_run_fail:2026-06-08": {
+    "timestamp": "2026-06-08T02:20:30.424757Z",
+    "library": "com.clickhouse:jdbc-v2:0.9.7",
+    "starting_commit": "d70f22852b59c7eb8783ccbc45fd636ace86111b",
+    "ending_commit": "003e82d9f6ffab7742034115eb4e4c2e7d780be6",
+    "previous_library": "com.clickhouse:jdbc-v2:0.9.5",
+    "strategy_name": "java_run_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "0.9.7",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 4,
+            "totalCalls": 4
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 4,
+        "totalCalls": 4
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 16547,
+          "missed": 110294,
+          "ratio": 0.130455,
+          "total": 126841
+        },
+        "line": {
+          "covered": 2801,
+          "missed": 23898,
+          "ratio": 0.10491,
+          "total": 26699
+        },
+        "method": {
+          "covered": 550,
+          "missed": 5995,
+          "ratio": 0.084034,
+          "total": 6545
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "0.9.5",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 4,
+            "totalCalls": 4
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 4,
+        "totalCalls": 4
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 14305,
+          "missed": 107842,
+          "ratio": 0.117113,
+          "total": 122147
+        },
+        "line": {
+          "covered": 2746,
+          "missed": 23546,
+          "ratio": 0.104442,
+          "total": 26292
+        },
+        "method": {
+          "covered": 539,
+          "missed": 5630,
+          "ratio": 0.087372,
+          "total": 6169
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 48836,
+      "cached_input_tokens_used": 516608,
+      "output_tokens_used": 2457,
+      "iterations": 2,
+      "cost_usd": 0.332,
+      "tested_library_loc": 2801,
+      "metadata_entries": 21,
+      "test_only_metadata_entries": 19,
+      "previous_library_metadata_entries": 23,
+      "previous_library_test_only_metadata_entries": 19,
+      "code_coverage_percent": 10.49,
+      "previous_library_coverage_percent": 10.44
+    },
+    "artifacts": {
+      "test_file": "tests/src/com.clickhouse/jdbc-v2/0.9.7/src/test/java/com_clickhouse/jdbc_v2/JdbcUtilsTest.java",
+      "metadata_file": "metadata/com.clickhouse/jdbc-v2/0.9.7/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/com.clickhouse/jdbc-v2/0.9.7/stats.json
+++ b/stats/com.clickhouse/jdbc-v2/0.9.7/stats.json
@@ -1,0 +1,37 @@
+{
+  "versions" : [ {
+    "version" : "0.9.7",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 4,
+          "totalCalls" : 4
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 4,
+      "totalCalls" : 4
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 16547,
+        "missed" : 110294,
+        "ratio" : 0.130455,
+        "total" : 126841
+      },
+      "line" : {
+        "covered" : 2801,
+        "missed" : 23898,
+        "ratio" : 0.10491,
+        "total" : 26699
+      },
+      "method" : {
+        "covered" : 550,
+        "missed" : 5995,
+        "ratio" : 0.084034,
+        "total" : 6545
+      }
+    }
+  } ]
+}

--- a/tests/src/com.clickhouse/jdbc-v2/0.9.7/.gitignore
+++ b/tests/src/com.clickhouse/jdbc-v2/0.9.7/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/com.clickhouse/jdbc-v2/0.9.7/build.gradle
+++ b/tests/src/com.clickhouse/jdbc-v2/0.9.7/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "com.clickhouse:jdbc-v2:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/com.clickhouse/jdbc-v2/0.9.7/gradle.properties
+++ b/tests/src/com.clickhouse/jdbc-v2/0.9.7/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = com.clickhouse:jdbc-v2:0.9.7
+library.version = 0.9.7
+metadata.dir = com.clickhouse/jdbc-v2/0.9.7/

--- a/tests/src/com.clickhouse/jdbc-v2/0.9.7/settings.gradle
+++ b/tests/src/com.clickhouse/jdbc-v2/0.9.7/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'com.clickhouse.jdbc-v2_tests'

--- a/tests/src/com.clickhouse/jdbc-v2/0.9.7/src/test/java/com_clickhouse/jdbc_v2/JdbcUtilsTest.java
+++ b/tests/src/com.clickhouse/jdbc-v2/0.9.7/src/test/java/com_clickhouse/jdbc_v2/JdbcUtilsTest.java
@@ -1,0 +1,46 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_clickhouse.jdbc_v2;
+
+import com.clickhouse.jdbc.internal.JdbcUtils;
+import org.junit.jupiter.api.Test;
+
+import java.sql.SQLException;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class JdbcUtilsTest {
+    @Test
+    void convertListCreatesNestedTypedArrays() throws SQLException {
+        List<?> values = List.of(
+                List.of(1, 2),
+                List.of(3, 4));
+
+        Object converted = JdbcUtils.convertList(values, String.class, 2);
+
+        assertThat(converted).isInstanceOf(String[][].class);
+        String[][] actual = (String[][]) converted;
+        assertThat(actual[0]).containsExactly("1", "2");
+        assertThat(actual[1]).containsExactly("3", "4");
+    }
+
+    @Test
+    void convertArrayCreatesNestedTypedArrays() throws SQLException {
+        Object values = new Object[] {
+                new Object[] {1, 2},
+                new Object[] {3, 4}
+        };
+
+        Object converted = JdbcUtils.convertArray(values, String.class, 2);
+
+        assertThat(converted).isInstanceOf(String[][].class);
+        String[][] actual = (String[][]) converted;
+        assertThat(actual[0]).containsExactly("1", "2");
+        assertThat(actual[1]).containsExactly("3", "4");
+    }
+}

--- a/tests/src/com.clickhouse/jdbc-v2/0.9.7/src/test/java/com_clickhouse/jdbc_v2/Jdbc_v2Test.java
+++ b/tests/src/com.clickhouse/jdbc-v2/0.9.7/src/test/java/com_clickhouse/jdbc_v2/Jdbc_v2Test.java
@@ -1,0 +1,381 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_clickhouse.jdbc_v2;
+
+import com.clickhouse.data.ClickHouseColumn;
+import com.clickhouse.data.ClickHouseDataType;
+import com.clickhouse.data.Tuple;
+import com.clickhouse.jdbc.DataSourceImpl;
+import com.clickhouse.jdbc.Driver;
+import com.clickhouse.jdbc.PreparedStatementImpl;
+import com.clickhouse.jdbc.metadata.ResultSetMetaDataImpl;
+import com.clickhouse.jdbc.types.Array;
+import org.junit.jupiter.api.Test;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.math.BigDecimal;
+import java.net.InetAddress;
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.DriverPropertyInfo;
+import java.sql.JDBCType;
+import java.sql.ParameterMetaData;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.sql.Statement;
+import java.sql.Types;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.TimeZone;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class Jdbc_v2Test {
+    private static final String URL = "jdbc:clickhouse://localhost:8123/default";
+
+    @Test
+    void driverRecognizesClickHouseUrlsAndExposesConfiguration() throws Exception {
+        Driver driver = new Driver();
+        Properties properties = offlineProperties();
+        properties.setProperty("user", "property-user");
+
+        assertThat(driver.acceptsURL("jdbc:clickhouse://localhost:8123/default")).isTrue();
+        assertThat(driver.acceptsURL("jdbc:ch://localhost:8123/default")).isTrue();
+        assertThat(driver.acceptsURL("jdbc:postgresql://localhost:5432/postgres")).isFalse();
+        assertThat(driver.connect("jdbc:postgresql://localhost:5432/postgres", properties)).isNull();
+        assertThat(driver.jdbcCompliant()).isFalse();
+        assertThat(driver.getMajorVersion()).isEqualTo(Driver.getDriverMajorVersion());
+        assertThat(driver.getMinorVersion()).isEqualTo(Driver.getDriverMinorVersion());
+        assertThat(Driver.chSettingKey("max_threads")).isEqualTo("clickhouse_setting_max_threads");
+        assertThatThrownBy(driver::getParentLogger).isInstanceOf(SQLFeatureNotSupportedException.class);
+
+        DriverPropertyInfo[] propertyInfos = driver.getPropertyInfo(
+                "jdbc:clickhouse://localhost:8123/analytics?user=url-user&socket_timeout=250", properties);
+        Map<String, String> valuesByName = new LinkedHashMap<>();
+        for (DriverPropertyInfo propertyInfo : propertyInfos) {
+            valuesByName.put(propertyInfo.name, propertyInfo.value);
+        }
+
+        assertThat(valuesByName)
+                .containsEntry("database", "analytics")
+                .containsEntry("user", "url-user")
+                .containsEntry("socket_timeout", "250")
+                .containsKey("server_time_zone");
+    }
+
+    @Test
+    void dataSourceProvidesJdbcWrapperSemanticsAndConnections() throws Exception {
+        DataSourceImpl dataSource = new DataSourceImpl(URL, offlineProperties());
+        StringWriter sink = new StringWriter();
+        PrintWriter logWriter = new PrintWriter(sink);
+
+        dataSource.setLogWriter(logWriter);
+
+        assertThat(dataSource.getLogWriter()).isSameAs(logWriter);
+        assertThat(dataSource.isWrapperFor(DataSourceImpl.class)).isTrue();
+        assertThat(dataSource.unwrap(DataSourceImpl.class)).isSameAs(dataSource);
+        assertThatThrownBy(() -> dataSource.unwrap(String.class)).isInstanceOf(SQLException.class);
+        assertThatThrownBy(() -> dataSource.setLoginTimeout(1)).isInstanceOf(SQLFeatureNotSupportedException.class);
+        assertThatThrownBy(dataSource::getLoginTimeout).isInstanceOf(SQLFeatureNotSupportedException.class);
+        assertThatThrownBy(dataSource::getParentLogger).isInstanceOf(SQLFeatureNotSupportedException.class);
+
+        try (Connection connection = dataSource.getConnection("user-from-call", "secret")) {
+            assertThat(connection).isNotNull();
+            assertThat(connection.isClosed()).isFalse();
+        }
+    }
+
+    @Test
+    void connectionExposesOfflineJdbcStateAndUnsupportedOperations() throws Exception {
+        try (Connection connection = openConnection()) {
+            assertThat(connection.getAutoCommit()).isTrue();
+            assertThat(connection.isReadOnly()).isFalse();
+            assertThat(connection.getTransactionIsolation()).isEqualTo(Connection.TRANSACTION_NONE);
+            assertThat(connection.isValid(0)).isFalse();
+            assertThat(connection.getSchema()).isEqualTo("default");
+            assertThat(connection.getCatalog()).isNull();
+            assertThat(connection.getWarnings() == null).isTrue();
+            assertThat(connection.getHoldability()).isEqualTo(ResultSet.HOLD_CURSORS_OVER_COMMIT);
+            assertThat(connection.isWrapperFor(Connection.class)).isTrue();
+            assertThat(connection.unwrap(Connection.class)).isSameAs(connection);
+
+            connection.setSchema("analytics");
+            connection.setClientInfo("ApplicationName", "metadata-test");
+            assertThat(connection.getSchema()).isEqualTo("analytics");
+            assertThat(connection.getClientInfo("ApplicationName")).isEqualTo("metadata-test");
+            assertThat(connection.getClientInfo()).containsEntry("ApplicationName", "metadata-test");
+
+            DatabaseMetaData metadata = connection.getMetaData();
+            assertThat(metadata.getDatabaseProductName()).isEqualTo("ClickHouse");
+            assertThat(metadata.getDriverName()).isEqualTo("ClickHouse JDBC Driver");
+            assertThat(metadata.supportsTransactions()).isFalse();
+            assertThat(metadata.supportsMultipleResultSets()).isFalse();
+            assertThat(metadata.supportsColumnAliasing()).isTrue();
+
+            assertThatThrownBy(() -> connection.setAutoCommit(false))
+                    .isInstanceOf(SQLFeatureNotSupportedException.class);
+            assertThatThrownBy(connection::commit).isInstanceOf(SQLFeatureNotSupportedException.class);
+            assertThatThrownBy(connection::rollback).isInstanceOf(SQLFeatureNotSupportedException.class);
+            connection.setReadOnly(true);
+            assertThat(connection.isReadOnly()).isTrue();
+            assertThatThrownBy(() -> connection.setTransactionIsolation(Connection.TRANSACTION_READ_COMMITTED))
+                    .isInstanceOf(SQLFeatureNotSupportedException.class);
+        }
+    }
+
+    @Test
+    void statementMaintainsJdbcStateWithoutExecutingQueries() throws Exception {
+        try (Connection connection = openConnection(); Statement statement = connection.createStatement()) {
+            assertThat(statement.getConnection()).isSameAs(connection);
+            assertThat(statement.getResultSet()).isNull();
+            assertThat(statement.getUpdateCount()).isEqualTo(-1);
+            assertThat(statement.getMoreResults()).isFalse();
+            assertThat(statement.getWarnings() == null).isTrue();
+            assertThat(statement.getFetchDirection()).isEqualTo(ResultSet.FETCH_FORWARD);
+            assertThat(statement.getFetchSize()).isPositive();
+            assertThat(statement.getResultSetConcurrency()).isEqualTo(ResultSet.CONCUR_READ_ONLY);
+            assertThat(statement.getResultSetType()).isEqualTo(ResultSet.TYPE_FORWARD_ONLY);
+            assertThat(statement.getResultSetHoldability()).isEqualTo(ResultSet.HOLD_CURSORS_OVER_COMMIT);
+            assertThat(statement.isPoolable()).isFalse();
+            assertThat(statement.isCloseOnCompletion()).isFalse();
+            assertThat(statement.getGeneratedKeys()).isNull();
+            assertThat(statement.getLargeUpdateCount()).isEqualTo(-1L);
+
+            statement.setMaxRows(42);
+            statement.setQueryTimeout(3);
+            statement.setFetchSize(128);
+            statement.setPoolable(true);
+            statement.addBatch("INSERT INTO t VALUES (1)");
+            statement.clearBatch();
+            statement.clearWarnings();
+            statement.closeOnCompletion();
+
+            assertThat(statement.getMaxRows()).isEqualTo(42);
+            assertThat(statement.getLargeMaxRows()).isEqualTo(42L);
+            assertThat(statement.getQueryTimeout()).isEqualTo(3);
+            assertThat(statement.getFetchSize()).isEqualTo(128);
+            assertThat(statement.isPoolable()).isTrue();
+            assertThat(statement.isWrapperFor(Statement.class)).isTrue();
+            assertThat(statement.unwrap(Statement.class)).isSameAs(statement);
+
+            statement.setFetchDirection(ResultSet.FETCH_REVERSE);
+            statement.setMaxFieldSize(100);
+            assertThat(statement.getFetchDirection()).isEqualTo(ResultSet.FETCH_FORWARD);
+            assertThat(statement.getMaxFieldSize()).isEqualTo(100);
+            assertThatThrownBy(() -> statement.setFetchDirection(-1)).isInstanceOf(SQLException.class);
+        }
+    }
+
+    @Test
+    void preparedStatementReportsParametersAndRejectsStatementSqlOverloads() throws Exception {
+        try (Connection connection = openConnection();
+                PreparedStatement statement = connection.prepareStatement("SELECT ? AS first, ? AS second")) {
+            ParameterMetaData parameterMetaData = statement.getParameterMetaData();
+
+            assertThat(parameterMetaData.getParameterCount()).isEqualTo(2);
+            assertThat(parameterMetaData.isNullable(1)).isEqualTo(ParameterMetaData.parameterNullableUnknown);
+            assertThat(parameterMetaData.isSigned(1)).isFalse();
+            assertThat(parameterMetaData.getPrecision(1)).isZero();
+            assertThat(parameterMetaData.getScale(1)).isZero();
+            assertThat(parameterMetaData.getParameterType(1)).isEqualTo(Types.OTHER);
+            assertThat(parameterMetaData.getParameterTypeName(1)).isEqualTo("UNKNOWN");
+            assertThat(parameterMetaData.getParameterClassName(1)).isEqualTo(Object.class.getName());
+            assertThat(parameterMetaData.getParameterMode(1)).isEqualTo(ParameterMetaData.parameterModeIn);
+            assertThatThrownBy(() -> parameterMetaData.getParameterType(0)).isInstanceOf(SQLException.class);
+            assertThatThrownBy(() -> parameterMetaData.getParameterType(3)).isInstanceOf(SQLException.class);
+
+            statement.setInt(1, 7);
+            statement.setString(2, "ClickHouse's JDBC");
+            statement.clearParameters();
+            statement.setObject(1, LocalDate.of(2024, 1, 2), JDBCType.DATE);
+            statement.setObject(2, Arrays.asList("alpha", "beta"));
+            assertThatThrownBy(() -> statement.setObject(2, Arrays.asList("gamma", "delta"), JDBCType.ARRAY))
+                    .isInstanceOf(SQLException.class)
+                    .hasMessageContaining("requires a parameter");
+            assertThat(statement.isPoolable()).isTrue();
+            assertThat(statement.isWrapperFor(PreparedStatement.class)).isTrue();
+            assertThat(statement.unwrap(PreparedStatement.class)).isSameAs(statement);
+
+            assertThatThrownBy(() -> statement.executeQuery("SELECT 1")).isInstanceOf(SQLException.class);
+            assertThatThrownBy(() -> statement.executeUpdate("SELECT 1")).isInstanceOf(SQLException.class);
+            assertThatThrownBy(() -> statement.execute("SELECT 1")).isInstanceOf(SQLException.class);
+            assertThatThrownBy(() -> statement.addBatch("SELECT 1")).isInstanceOf(SQLException.class);
+            assertThatThrownBy(() -> statement.setRowId(1, null)).isInstanceOf(SQLException.class);
+            assertThatThrownBy(() -> statement.setRef(1, null)).isInstanceOf(SQLFeatureNotSupportedException.class);
+        }
+    }
+
+    @Test
+    void preparedStatementAcceptsCommonJavaValueTypes() throws Exception {
+        try (Connection connection = openConnection();
+                PreparedStatement statement = connection.prepareStatement(
+                        "INSERT INTO t VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)")) {
+            Map<String, Integer> map = new LinkedHashMap<>();
+            map.put("one", 1);
+            map.put("two", 2);
+
+            statement.setBoolean(1, true);
+            statement.setBigDecimal(2, new BigDecimal("123.45"));
+            statement.setBytes(3, new byte[] {1, 2, 3});
+            statement.setDate(4, java.sql.Date.valueOf(LocalDate.of(2024, 2, 3)));
+            statement.setTime(5, java.sql.Time.valueOf(LocalTime.of(4, 5, 6)));
+            statement.setTimestamp(6, java.sql.Timestamp.valueOf(LocalDateTime.of(2024, 2, 3, 4, 5, 6, 7)));
+            statement.setObject(7, OffsetDateTime.parse("2024-02-03T04:05:06Z"));
+            statement.setObject(8, Instant.parse("2024-02-03T04:05:06Z"));
+            statement.setObject(9, InetAddress.getByName("127.0.0.1"));
+            statement.setArray(10, stringArray("x", "y"));
+            statement.setObject(11, map);
+            statement.setNull(12, Types.NULL);
+            statement.addBatch();
+        }
+    }
+
+    @Test
+    void connectionCreatesJdbcArraysFromClickHouseTypeNames() throws Exception {
+        try (Connection connection = openConnection()) {
+            java.sql.Array stringArray = connection.createArrayOf("String", new Object[] {"alpha", "beta"});
+            java.sql.Array integerArray = connection.createArrayOf("Int32", new Object[] {1, 2, 3});
+            java.sql.Array emptyArray = connection.createArrayOf("Bool", new Object[0]);
+            try {
+                assertThat(stringArray.getBaseTypeName()).isEqualTo("String");
+                assertThat(stringArray.getBaseType()).isEqualTo(Types.VARCHAR);
+                assertThat((Object[]) stringArray.getArray()).containsExactly("alpha", "beta");
+
+                assertThat(integerArray.getBaseTypeName()).isEqualTo("Int32");
+                assertThat(integerArray.getBaseType()).isEqualTo(Types.INTEGER);
+                assertThat((Object[]) integerArray.getArray()).containsExactly(1, 2, 3);
+
+                assertThat(emptyArray.getBaseTypeName()).isEqualTo("Bool");
+                assertThat(emptyArray.getBaseType()).isEqualTo(Types.BOOLEAN);
+                assertThat((Object[]) emptyArray.getArray()).isEmpty();
+
+                assertThatThrownBy(() -> connection.createArrayOf("NotAClickHouseType", new Object[] {"value"}))
+                        .isInstanceOf(IllegalArgumentException.class);
+            } finally {
+                stringArray.free();
+                integerArray.free();
+                emptyArray.free();
+            }
+        }
+    }
+
+    @Test
+    void resultSetMetadataDescribesClickHouseColumnsAsJdbcColumns() throws Exception {
+        List<ClickHouseColumn> columns = List.of(
+                ClickHouseColumn.of("id", "Int32"),
+                ClickHouseColumn.of("amount", "Nullable(Decimal(12, 2))"),
+                ClickHouseColumn.of("tags", "Array(String)"));
+        Map<ClickHouseDataType, Class<?>> typeClassMap = new LinkedHashMap<>();
+        typeClassMap.put(ClickHouseDataType.Int32, Integer.class);
+        typeClassMap.put(ClickHouseDataType.Decimal, BigDecimal.class);
+        typeClassMap.put(ClickHouseDataType.Array, Object[].class);
+
+        ResultSetMetaData metadata = new ResultSetMetaDataImpl(
+                columns, "analytics", "local", "events", typeClassMap);
+
+        assertThat(metadata.getColumnCount()).isEqualTo(3);
+        assertThat(metadata.getColumnName(1)).isEqualTo("id");
+        assertThat(metadata.getColumnLabel(1)).isEqualTo("id");
+        assertThat(metadata.getColumnType(1)).isEqualTo(Types.INTEGER);
+        assertThat(metadata.getColumnTypeName(1)).isEqualTo("Int32");
+        assertThat(metadata.isSigned(1)).isTrue();
+        assertThat(metadata.isNullable(1)).isEqualTo(ResultSetMetaData.columnNoNulls);
+        assertThat(metadata.getColumnClassName(1)).isEqualTo(Integer.class.getName());
+
+        assertThat(metadata.getSchemaName(2)).isEqualTo("analytics");
+        assertThat(metadata.getCatalogName(2)).isEqualTo("local");
+        assertThat(metadata.getTableName(2)).isEqualTo("events");
+        assertThat(metadata.getColumnType(2)).isEqualTo(Types.DECIMAL);
+        assertThat(metadata.getColumnTypeName(2)).contains("Decimal");
+        assertThat(metadata.getPrecision(2)).isEqualTo(12);
+        assertThat(metadata.getScale(2)).isEqualTo(2);
+        assertThat(metadata.isNullable(2)).isEqualTo(ResultSetMetaData.columnNullable);
+        assertThat(metadata.getColumnClassName(2)).isEqualTo(BigDecimal.class.getName());
+
+        assertThat(metadata.getColumnType(3)).isEqualTo(Types.ARRAY);
+        assertThat(metadata.getColumnTypeName(3)).isEqualTo("Array(String)");
+        assertThat(metadata.getColumnClassName(3)).isEqualTo(Object[].class.getName());
+        assertThat(metadata.isAutoIncrement(3)).isFalse();
+        assertThat(metadata.isReadOnly(3)).isTrue();
+        assertThat(metadata.isWritable(3)).isFalse();
+        assertThatThrownBy(() -> metadata.getColumnName(0)).isInstanceOf(SQLException.class);
+    }
+
+    @Test
+    void clickHouseArrayAndTupleExposeValueObjects() throws Exception {
+        Array array = stringArray("one", "two", "three");
+        Tuple tuple = new Tuple("name", 42, true);
+
+        assertThat(array.getBaseTypeName()).isEqualTo("String");
+        assertThat(array.getBaseType()).isEqualTo(Types.VARCHAR);
+        assertThat((Object[]) array.getArray()).containsExactly("one", "two", "three");
+        assertThat((Object[]) array.getArray(1, 2)).containsExactly("two", "three");
+        try (ResultSet resultSet = array.getResultSet()) {
+            ResultSetMetaData metadata = resultSet.getMetaData();
+
+            assertThat(metadata.getColumnCount()).isEqualTo(2);
+            assertThat(metadata.getColumnName(1)).isEqualTo("INDEX");
+            assertThat(metadata.getColumnName(2)).isEqualTo("VALUE");
+            assertThat(resultSet.isBeforeFirst()).isTrue();
+            assertThat(resultSet.next()).isTrue();
+            assertThat(resultSet.getInt("INDEX")).isEqualTo(1);
+            assertThat(resultSet.getString("VALUE")).isEqualTo("one");
+            assertThat(resultSet.next()).isTrue();
+            assertThat(resultSet.getObject(1)).isEqualTo(2);
+            assertThat(resultSet.getObject(2, String.class)).isEqualTo("two");
+        }
+        assertThatThrownBy(() -> array.getArray(Map.of())).isInstanceOf(SQLFeatureNotSupportedException.class);
+        assertThatThrownBy(() -> array.getResultSet(Map.of())).isInstanceOf(SQLFeatureNotSupportedException.class);
+        assertThatThrownBy(() -> array.getArray(-1, 1)).isInstanceOf(SQLException.class);
+        array.free();
+        assertThatThrownBy(array::getArray).isInstanceOf(SQLException.class);
+
+        assertThat(tuple.size()).isEqualTo(3);
+        assertThat(tuple.getValue(0)).isEqualTo("name");
+        assertThat(tuple.getValues()).containsExactly("name", 42, true);
+        assertThat(tuple).hasToString("(name, 42, true)");
+    }
+
+    @Test
+    void preparedStatementUtilityReplacesOnlyUnquotedQuestionMarks() {
+        String sql = "SELECT ?, '?', \"?\", `?`, col FROM table WHERE id = ?";
+
+        assertThat(PreparedStatementImpl.replaceQuestionMarks(sql, "NULL"))
+                .isEqualTo("SELECT NULL, '?', \"?\", `?`, col FROM table WHERE id = NULL");
+    }
+
+    private static Connection openConnection() throws SQLException {
+        return new Driver().connect(URL, offlineProperties());
+    }
+
+    private static Array stringArray(Object... elements) throws SQLException {
+        return new Array(ClickHouseColumn.of("values", "Array(String)"), elements);
+    }
+
+    private static Properties offlineProperties() {
+        Properties properties = new Properties();
+        properties.setProperty("server_time_zone", TimeZone.getTimeZone("UTC").getID());
+        properties.setProperty("disable_frameworks_detection", "true");
+        properties.setProperty("connection_pool_enabled", "false");
+        properties.setProperty("user", "default");
+        properties.setProperty("password", "");
+        return properties;
+    }
+}

--- a/tests/src/com.clickhouse/jdbc-v2/0.9.7/src/test/java/com_clickhouse/jdbc_v2/Jdbc_v2Test.java
+++ b/tests/src/com.clickhouse/jdbc-v2/0.9.7/src/test/java/com_clickhouse/jdbc_v2/Jdbc_v2Test.java
@@ -45,6 +45,7 @@ import java.util.Properties;
 import java.util.TimeZone;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 public class Jdbc_v2Test {
@@ -206,9 +207,8 @@ public class Jdbc_v2Test {
             statement.clearParameters();
             statement.setObject(1, LocalDate.of(2024, 1, 2), JDBCType.DATE);
             statement.setObject(2, Arrays.asList("alpha", "beta"));
-            assertThatThrownBy(() -> statement.setObject(2, Arrays.asList("gamma", "delta"), JDBCType.ARRAY))
-                    .isInstanceOf(SQLException.class)
-                    .hasMessageContaining("requires a parameter");
+            assertThatCode(() -> statement.setObject(2, Arrays.asList("gamma", "delta"), JDBCType.ARRAY))
+                    .doesNotThrowAnyException();
             assertThat(statement.isPoolable()).isTrue();
             assertThat(statement.isWrapperFor(PreparedStatement.class)).isTrue();
             assertThat(statement.unwrap(PreparedStatement.class)).isSameAs(statement);
@@ -372,7 +372,6 @@ public class Jdbc_v2Test {
     private static Properties offlineProperties() {
         Properties properties = new Properties();
         properties.setProperty("server_time_zone", TimeZone.getTimeZone("UTC").getID());
-        properties.setProperty("disable_frameworks_detection", "true");
         properties.setProperty("connection_pool_enabled", "false");
         properties.setProperty("user", "default");
         properties.setProperty("password", "");

--- a/tests/src/com.clickhouse/jdbc-v2/0.9.7/src/test/resources/META-INF/native-image/com.clickhouse/jdbc-v2-tests/native-image.properties
+++ b/tests/src/com.clickhouse/jdbc-v2/0.9.7/src/test/resources/META-INF/native-image/com.clickhouse/jdbc-v2-tests/native-image.properties
@@ -1,0 +1,1 @@
+Args = --enable-url-protocols=http

--- a/tests/src/com.clickhouse/jdbc-v2/0.9.7/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/com.clickhouse/jdbc-v2/0.9.7/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -1,0 +1,115 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "com_clickhouse.jdbc_v2.Jdbc_v2Test"
+      },
+      "type" : "java.lang.OutOfMemoryError",
+      "jniAccessible" : true
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_clickhouse.jdbc_v2.Jdbc_v2Test"
+      },
+      "type" : "net.jpountz.lz4.LZ4HCJNICompressor",
+      "fields" : [
+        {
+          "name" : "INSTANCE"
+        }
+      ],
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [
+            "int"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_clickhouse.jdbc_v2.Jdbc_v2Test"
+      },
+      "type" : "net.jpountz.lz4.LZ4JNICompressor",
+      "fields" : [
+        {
+          "name" : "INSTANCE"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_clickhouse.jdbc_v2.Jdbc_v2Test"
+      },
+      "type" : "net.jpountz.lz4.LZ4JNIFastDecompressor",
+      "fields" : [
+        {
+          "name" : "INSTANCE"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_clickhouse.jdbc_v2.Jdbc_v2Test"
+      },
+      "type" : "net.jpountz.lz4.LZ4JNISafeDecompressor",
+      "fields" : [
+        {
+          "name" : "INSTANCE"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_clickhouse.jdbc_v2.Jdbc_v2Test"
+      },
+      "type" : "sun.security.provider.NativePRNG",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [
+            "java.security.SecureRandomParameters"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_clickhouse.jdbc_v2.Jdbc_v2Test"
+      },
+      "type" : "sun.security.provider.SHA",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    }
+  ],
+  "resources" : [
+    {
+      "condition" : {
+        "typeReached" : "com_clickhouse.jdbc_v2.Jdbc_v2Test"
+      },
+      "glob" : "META-INF/services/org.assertj.core.configuration.Configuration"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_clickhouse.jdbc_v2.Jdbc_v2Test"
+      },
+      "glob" : "META-INF/services/org.assertj.core.presentation.Representation"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_clickhouse.jdbc_v2.Jdbc_v2Test"
+      },
+      "glob" : "net/jpountz/util/linux/amd64/liblz4-java.so"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_clickhouse.jdbc_v2.Jdbc_v2Test"
+      },
+      "glob" : "org/publicsuffix/list/effective_tld_names.dat"
+    }
+  ]
+}

--- a/tests/src/com.clickhouse/jdbc-v2/0.9.7/user-code-filter.json
+++ b/tests/src/com.clickhouse/jdbc-v2/0.9.7/user-code-filter.json
@@ -1,0 +1,13 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "com.clickhouse.**"
+    },
+    {
+      "includeClasses" : "com_clickhouse.jdbc_v2.**"
+    }
+  ]
+}

